src/app/organisms/emoji-board/custom-emoji.js

```diff
@@ -0,0 +1,74 @@
+import { emojis } from './emoji';
+
+// Custom emoji are stored in one of three places:
+// - User emojis, which are stored in account data
+// - Room emojis, which are stored in state events in a room
+// - Emoji packs, which are rooms of emojis referenced in the account data or in a room's
+//   cannonical space
+//
+// Emojis and packs referenced from within a user's account data should be available
+// globally, while emojis and packs in rooms and spaces should only be available within
+// those spaces and rooms
+
+// Retrieve a list of user emojis
+//
+// Result is a list of objects, each with a shortcode and an mxc property
+//
+// Accepts a reference to a matrix client as the only argument
+function getUserEmoji(mx) {
+  const accountDataEmoji = mx.getAccountData('im.ponies.user_emotes');
+  if (!accountDataEmoji) {
+    return [];
+  }
+
+  const { images } = accountDataEmoji.event.content;
+  const mapped = Object.entries(images).map((e) => ({
+    shortcode: e[0],
+    mxc: e[1].url,
+  }));
+  return mapped;
+}
+
+// Returns all user emojis and all standard unicode emojis
+//
+// Accepts a reference to a matrix client as the only argument
+//
+// Result is a map from shortcode to the corresponding emoji.  If two emoji share a
+// shortcode, only one will be presented, with priority given to custom emoji.
+//
+// Will eventually be expanded to include all emojis revelant to a room and the user
+function getAllEmoji(mx) {
+  const allEmoji = new Map();
+
+  for (let i = 0; i < emojis.length; i += 1) {
+    const emoji = emojis[i];
+    for (let j = 0; j < emoji.shortcodes; j += 1) {
+      allEmoji.set(emoji.shortcodes[j], emoji);
+    }
+  }
+
+  getUserEmoji(mx).forEach((emoji) => {
+    allEmoji.set(emoji.shortcode, emoji);
+  });
+
+  return allEmoji;
+}
```

Comment:
This function should be called getShortcodeToEmoji as it doesn't return emojis but a mapping of shortcode.

Also the for loop at line 45 consider `emoji.shortcodes` as an array but it is an array only if there is more then 1 shortcode otherwise it's a string.

You forget to write emoji.shortcodes`.length` in line 45 loop. Recommended forEach instead :D